runtime/calamari/src/lib.rs

```diff
@@ -1023,10 +1023,33 @@ impl_runtime_apis! {
 
     impl nimbus_primitives::NimbusApi<Block> for Runtime {
         fn can_author(author: NimbusId, relay_parent: u32, parent_header: &<Block as BlockT>::Header) -> bool {
-            System::initialize(&(parent_header.number + 1), &parent_header.hash(), &parent_header.digest);
-
-            // And now the actual prediction call
-            <AuthorInherent as nimbus_primitives::CanAuthor<_>>::can_author(&author, &relay_parent)
+            let next_block_number = parent_header.number + 1;
+            let slot = relay_parent;
+            // Because the staking solution calculates the next staking set at the beginning
+            // of the first block in the new round, the only way to accurately predict the
+            // authors is to compute the selection during prediction.
+            // NOTE: This logic must manually be kept in sync with the nimbus filter pipeline
+            if pallet_parachain_staking::Pallet::<Self>::round().should_update(next_block_number)
+            {
+                // lookup account from nimbusId
+                // mirrors logic in `pallet_author_inherent`
+                use nimbus_primitives::AccountLookup;
+                let account = match manta_collator_selection::Pallet::<Self>::lookup_account(&author) {
+                    Some(account) => account,
+                    // Authors whose account lookups fail will not be eligible
+                    None => {
+                        return false;
+                    }
+                };
+                // manually check aura eligibility (in the new round)
+                // mirrors logic in `aura_style_filter`
+                let truncated_half_slot = (slot >> 1) as usize;
+                let active: Vec<AccountId> = pallet_parachain_staking::Pallet::<Self>::compute_top_candidates();
```

Comment:
`aura_style_filter` is using storage getter: `selected_candidates`. while here compute again, is that any other reason we can't use `selected_candidates` here?